core/admin/port_management.py

```
@@ -0,0 +1,449 @@
+
+def close_monitoring_port(service: str, username: str) -> Dict:
+    """Close an open monitoring port"""
+    if service not in active_port_mappings:
+        return {"error": "Port not open", "service": service}
+    
+    mapping = active_port_mappings[service]
+    external_port = mapping["external_port"]
+    
+    # Remove iptables rules
+    try:
+        subprocess.run([
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)